From 2c04fa759de4dba003179b6d9d5c2cc6075af016 Mon Sep 17 00:00:00 2001
From: shellyxx <shellyx.xie@intel.com>
Date: Wed, 18 Dec 2019 14:36:40 +0800
Subject: [PATCH] Add "persist.sys.statusbar.hidden" to control the statusbar

Set persist.sys.statusbar.hidden to true, when need hidden the statusbar.

Tracked-On: OAM-88837
Signed-off-by: Xie, ShellyX <shellyx.xie@intel.com>
Change-Id: I3b5d843d4bebb611dd4f50eef0151aaaa40c761c
---
 .../src/com/android/systemui/statusbar/phone/StatusBar.java  | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index d6a770f75e2..4cdd9c5dd18 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -806,6 +806,11 @@ public class StatusBar extends SystemUI implements DemoMode,
         updateTheme();
 
         inflateStatusBarWindow(context);
+
+        // set persist.sys.statusbar.hidden to true, when need to hidden the statusbar.
+        if ("true".equals(SystemProperties.get("persist.sys.statusbar.hidden"))) {
+             mStatusBarWindow.setVisibility(View.INVISIBLE);
+        }
         mStatusBarWindow.setService(this);
         mStatusBarWindow.setOnTouchListener(getStatusBarWindowTouchListener());
 
-- 
2.17.1

